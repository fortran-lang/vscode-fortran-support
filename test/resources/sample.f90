program main
 ! execution
  
  call say_hello()

contains
  function sum(a, b)
    integer :: a,b
    integer :: sum
    sum = a + b
  end function sum

  integer function mul(a, b)
    integer :: a,b
    sum = a * b
  end function mul
  
  function sub(a, b) result(s)
    integer :: a,b
    integer :: s
    s = a - b
  end function sub

  subroutine say_hello(a,b)
    integer :: a,b

    print *, "Hello, World!"
  end subroutine say_hello

  pure integer function div(a, b)
    integer :: a,b
    div = a / b
  end function div

  pure function pow(a, b) result(e)
    integer :: a,b
    integer :: e
    e = a**b
  end function pow
end program main