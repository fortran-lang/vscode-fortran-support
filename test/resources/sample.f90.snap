>program main
#^^^^^^^ source.fortran.free meta.program.fortran keyword.control.program.fortran
#       ^ source.fortran.free meta.program.fortran
#        ^^^^ source.fortran.free meta.program.fortran entity.name.program.fortran
> ! execution
#^ source.fortran.free meta.program.fortran meta.block.specification.fortran
# ^ source.fortran.free meta.program.fortran meta.block.specification.fortran comment.line.fortran
#  ^^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran comment.line.fortran
>  
>  call say_hello()
#^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran
#  ^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran keyword.control.call.fortran
#      ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran
#       ^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran entity.name.function.subroutine.fortran
#                ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran punctuation.parentheses.left.fortran
#                 ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.statement.control.call.fortran punctuation.parentheses.right.fortran
>
>contains
#^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran keyword.control.contains.fortran
>
>  subroutine say_hello(a,b)
#^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.attribute-list.fortran
#  ^^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran keyword.other.subroutine.fortran
#            ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran
#             ^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran entity.name.function.subroutine.fortran
#                      ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.first-line.fortran punctuation.definition.parameters.begin.fortran
#                       ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.first-line.fortran variable.parameter.fortran
#                        ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.first-line.fortran
#                         ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.first-line.fortran variable.parameter.fortran
#                          ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.first-line.fortran punctuation.definition.parameters.end.fortran
>    integer :: a,b
#^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran
#    ^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran storage.type.integer.fortran
#           ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran meta.attribute-list.fortran
#            ^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran keyword.operator.double-colon.fortran
#              ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran meta.name-list.fortran
#               ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran meta.name-list.fortran meta.parameter.fortran
#                ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran meta.name-list.fortran
#                 ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran meta.specification.type.fortran meta.name-list.fortran meta.parameter.fortran
>
>    print *, "Hello, World!"
#^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran
#    ^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran keyword.control.print.fortran
#         ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran
#          ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran keyword.operator.power.fortran
#           ^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran
#             ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran string.quoted.double.fortran punctuation.definition.string.begin.fortran
#              ^^^^^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran string.quoted.double.fortran
#                           ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran string.quoted.double.fortran punctuation.definition.string.end.fortran
>  end subroutine say_hello
#^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran meta.block.specification.fortran
#  ^^^^^^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran keyword.other.endsubroutine.fortran
#                ^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran
#                 ^^^^^^^^^ source.fortran.free meta.program.fortran meta.block.specification.fortran meta.block.contains.fortran meta.subroutine.fortran entity.name.function.subroutine.fortran
>  
>  
>end program main
#^^^^^^^^^^^ source.fortran.free meta.program.fortran keyword.control.endprogram.fortran
#           ^ source.fortran.free meta.program.fortran
#            ^^^^ source.fortran.free meta.program.fortran entity.name.program.fortran